config_parser_test.cc

```diff
@@ -9,3 +9,34 @@ TEST(NginxConfigParserTest, SimpleConfig) {
 
   EXPECT_TRUE(success);
 }
+
+//Fixture implemented
+class NginxStringConfigParserTest : public ::testing::Test{
+protected:
+bool ParseString (const std::string config_string){
+  std:: stringstream config_stream(config_string);
+  return parser_.Parse(&config_stream, &out_config_);
+}
+NginxConfigParser parser_;
+NginxConfig out_config_;
+};
+
+TEST_F(NginxStringConfigParserTest, NotEmpty){
+  EXPECT_FALSE(ParseString(""));
+}
+
+TEST_F(NginxStringConfigParserTest, MissingNestedParanthesis){
+  EXPECT_FALSE(ParseString("fambli {fubli fibli;"));
```

Comment:
I think your fix handles it, but you could add a test for invalid nested brackets, like `fambli { fubli { fibli } frobly;`